env_spec.py

```
@@ -19,138 +28,216 @@
 
-    for line in lines:
-        try:
-            input_field, input_type = line.split(": ")
+            choices_match = re.match(choices_regex, line)
+            default_value_match = re.match(default_values_regex, line)
 
-            if check_input(input_field) and check_type(input_type):
-                html_output += render_env_var_spec(input_field, input_type)
-            else:
-                return ""
-        except:
-            input_field = line
+            if choices_match:
+                choices_str = choices_match.groups()[0]
 
-            if check_input(input_field):
-                html_output += render_env_var_spec(input_field, "")
+                choices = create_list(choices_str)
+
+                if not choices:
+                    try:
+                        raise EnvSpecSyntaxError
+                    except EnvSpecSyntaxError:
+                        return []
+                line_dict["choices"] = choices
+            else:
+                line_dict["choices"] = None
+
+            if default_value_match:
+                default_value = default_value_match.groups()[1]
+                default_value = default_value.strip()
+
+                if choices_match:
+                    if default_value not in choices or default_value == "":
+                        try:
+                            raise EnvSpecSyntaxError
+                        except EnvSpecSyntaxError:
+                            return []
+                line_dict["default_value"] = default_value
+            else:
+                line_dict["default_value"] = None
+
+            if not choices_match:
+                if default_value_match:
+                    type_t = default_value_match.groups()[0]
+                else:
+                    type_t = line
+
+                type_t = type_t.strip()
+
+                if type_t not in valid_types_list:
+                    try:
+                        raise EnvSpecSyntaxError
```

Can you please raise this exception along with a useful human friendly error?

```py
raise EnvSpecSyntaxError(‘this went wrong and that...`)
```